File: arq-base/computer.py
import ufc2x as cpu
import memory as mem
import clock as clk

mem.write_word(50, 21) 
mem.write_word(100, 32)
mem.write_word(130, 10)

# X <- X + memory[50]
mem.write_byte(1, 2)      # X <- X + memory...
mem.write_byte(2, 50)     # ...[50]

# if X=0 goto 7
mem.write_byte(3, 15)
mem.write_byte(4, 7)

# X <- X + memory[100]
mem.write_byte(5, 2)      # X <- X + memory...
mem.write_byte(6, 100)    # ...[100]

# X <- X - memory[130]
mem.write_byte(7, 6)
mem.write_byte(8, 130)
    
# memory[150] = X
mem.write_byte(9, 10)
mem.write_byte(10, 150)    
    
# stop
mem.write_byte(11, 255)


clk.start([cpu])

print(mem.read_word(150))

File: computer.cpp
#include <iostream>
#include <fstream>
#include "src/inc/memory.hpp"
#include "src/inc/cpu.hpp"
#include "examples/firmware3.hpp"

using namespace std;

int main (int argc, char **argv){
    if (argc <= 1){
        cout << "Memória não informada, sem instruções para executar\n";
        return EXIT_FAILURE;
    }

    MainMemory men;
	Cpu cpu;
    cpu.setMemory(&men);
    Word *firmware = get_firmware();
    cpu.setFirmware(firmware);

    ifstream input (argv[1], ios::in);

    int i = 1;
    men.write_byte (0, 0x00);

    while ( !input.eof() ){
        men.write_byte (i, input.get());
        ++i;
    }

    //men.print_memory(10);

    int time = cpu.start();
    
    cout << "word 1 da memória: " << men.read_word(1) << '\n';
    cout << "word 2 da memória: " << men.read_word(2) << '\n';

	cout << "Execução finalizada em: " << time << " passos\n\n";

    //men.print_memory(8);
    
    delete[] firmware;
    return EXIT_SUCCESS;
}

